lib/rubocop/ext/regexp_node.rb

```diff
@@ -54,17 +54,22 @@ def each_capture(named: ANY)
         return enum_for(__method__, named: named) unless block_given?
 
         parsed_tree&.traverse do |event, exp, _index|
-          yield(exp) if event == :enter &&
-                        named == exp.respond_to?(:name) &&
-                        exp.respond_to?(:capturing?) &&
-                        exp.capturing?
+          yield(exp) if exp_conditions(exp, event, named)
         end
 
         self
       end
 
       private
 
+      def exp_conditions(exp, event, named)
```

Comment:
Given that's a predicate method is should probably have a name looking like a predicate. "What are we checking for here exactly?" should be the principle you apply when picking the optimal name.

Reply:
Rename it with named_capturing.